index.html

```diff
@@ -3057,26 +3050,27 @@ <h3>Zero-Knowledge Proofs</h3>
 The next example utilizes the <a>verifiable credential</a> above to generate a
 new derived <a>verifiable credential</a> with a privacy-preserving proof. The
 derived <a>verifiable credential</a> is then placed in a
-<a>verifiable presentation</a>, which further proves that the entire assertion
-is valid. There are three requirements of most <a>verifiable presentations</a>
-when they are to be used in zero-knowledge systems:
+<a>verifiable presentation</a>, so that the <a>verifiable credential</a>
+discloses only the intended <a>claims</a> and additional credential
+metadata that the <a>holder</a> intended. In order to do this all of the
+following requirements are expected be met:
         </p>
 
         <ul>
           <li>
-Each derived <a>verifiable credential</a> within a
-<a>verifiable presentation</a> MUST have a <code>credentialSchema</code>
-<a>property</a>. This allows the derived <a>verifiable credential</a> to
-reference the credential definition used to generate the derived proof.
+Each derived <a>verifiable credential</a> within a <a>verifiable
+presentation</a> MUST contain all necessary information to verify the
+<a>verifiable credential</a> either by including it directly within the
+credential or by referencing the necessary information.
           </li>
           <li>
 A <a>verifiable presentation</a> MUST NOT leak information that would enable
 the <a>verifier</a> to correlate the <a>holder</a> across multiple
 <a>verifiable presentations</a>.
           </li>
           <li>
-The <a>verifiable presentation</a> MUST contain a <code>proof</code>
-<a>property</a> to enable the <a>verifier</a> to ascertain that all derived
+The <a>verifiable presentation</a> SHOULD contain a <code>proof</code>
+<a>property</a> to enable the <a>verifier</a> to check that all derived
 <a>verifiable credentials</a> in the <a>verifiable presentation</a> were issued
 to the same <a>holder</a> without leaking personally identifiable information
```

Comment:
```suggestion
<a>verifiable credentials</a> in the <a>verifiable presentation</a> were issued
to the same <a>holder</a> without leaking personally identifiable information
```
Somehow I missed this in previous reviews.

* Why do the `derived <a>verifiable credentials</a>` have to have been `issued
to the same <a>holder</a>`?
* Are `derived <a>verifiable credentials</a>` issued at all?
* Why are we checking whether the current holder was the original holder of any  `<a>verifiable credentials</a>`?

Reply:
Indy anoncreds zkp-credentials, which were used as the exemplar for much of the material in this section, require that issued credentials be bound to a holder. One reason for this is that anoncreds aren't bound to a subject, but are rather issued to the subject, so binding to the holder is a way to accomplish connection of the credential to the subject. Another is to give verifiers more confidence that credentials are linked even when the disclosed claims are kept to a minimum. Binding to a holder also makes sharing credentials more detectable and unwieldy, which is seen as valuable in the anoncreds ecosystem.

That said, there are more zkp-credential schemes now than there were when the section was written, and not all of them require holder binding, even though many still consider that best practice. This is why the MUST is being changed to a SHOULD.

Reply:
This suggestion appears to be the same as the current text in there. I presume this review comment was more about raising the questions which I think @brentzundel addressed. Please let me know if there's any additional changes that I missed to resolve this comment.

Reply:
> this review comment was more about raising the questions

Correct. However, I do not think that the answers provided by @brentzundel remove my concerns with this section.

I am mostly concerned about the "bound"/"binding" terminology in use here, which I think is a significantly stronger relationship than is actually present between these VCs and their Subjects *or* Holders. I think it would be better to say that xyz property/ies in the Credentials under specific discussion are (or SHOULD be or MUST be) set to strings or URIs which identify the Holder who is expected to Present the VC to any Verifiers downstream. Then, the Verifier is expected to check that _**either**_ some other qrs property/ies were (or SHOULD have been or MUST have been) set to strings or URIs which identify the Subject, ***OR*** the Subject is always the same as the Holder identified by the values of xyz property/ies, ... Or something along those lines.

I think that "anoncreds aren't bound to a subject, but are rather issued to the subject, so binding to the holder is a way to accomplish connection of the credential to the subject" translates roughly to "anoncreds are Issued to their Subject, which is defined as a sapient being, and which is not identified as the Subject within the VC. Identifying the Holder is an indirect way to accomplish association of the VC with its Subject," which Subject is not really obscured by this method, since the identified Holder is always the Subject and anyone who can access the value of the field identifying the Holder is thereby able to access to identifier of the Subject. Leading to another concern with this section, since it seems that the Subject anonymity desired to be delivered through ZKP, at least in part by identifying one Holder who must be the Subject, instead of identifying the Subject, isn't actually delivered. 

I'm sure I'm missing something exceedingly clever, and I hope someone can explain it in terms I can understand and that we can swiftly adopt (and/or edit slighly to adopt) for this section.

Reply:
> I am mostly concerned about the "bound"/"binding" terminology in use here, which I think is a significantly stronger relationship than is actually present between these VCs and their Subjects or Holders. I think it would be better to say that xyz property/ies in the Credentials under specific discussion are (or SHOULD be or MUST be) set to strings or URIs which identify the Holder who is expected to Present the VC to any Verifiers downstream.

+1. And if there are cases where there aren't actually any properties in the VC itself with this information because some mechanisms include it via secrets and/or proof data, then that can be stated as an option as well. I think it's important that we indicate that we're talking about setting expectations -- and not assert a level of control that isn't possible. It's also fine to say that there are mechanisms that attempt to impede adversaries that try to subvert these expectations, but such statements should include a friendly warning that they all have limitations that need to be well understood by implementers (and readers should consult external documentation for that).

Reply:
This thread continues in #821.

Reply:
Do we want to action this within this PR - if so can you propose some suggested changes based on the discussions so far.

Reply:
@kdenhartog -- There are no changes to make here in this PR based on this comment.